jax2onnx/converter/ir_optimizations.py

```diff
@@ -0,0 +1,1741 @@
+                ins[i] = new_name if new_name is not None else ""
+                changed = True
+        if changed:
+            if hasattr(m, "replace_input_with"):
+                try:
+                    for idx, iv in enumerate(ins):
+                        if iv is new_v:
+                            m.replace_input_with(idx, new_v)
+                except Exception:
+                    pass
+            try:
+                _set_node_inputs(m, ins)
+            except Exception:
+                # Fallback for pure string inputs
+                names = []
+                for iv in ins:
+                    if isinstance(iv, str):
+                        names.append(iv)
+                    else:
+                        nm = _v_name(iv)
+                        names.append(nm if nm is not None else "")
+                try:
+                    m.input = tuple(names)
+                except Exception:
+                    m.input = list(names)
+
+
+def _replace_in_graph_outputs(
+    graph, old_v: Optional["ir.Value"], old_name: Optional[str], new_v: "ir.Value"
+) -> None:
+    for attr in ("outputs", "output"):
+        outs = getattr(graph, attr, None)
+        if outs is None:
+            continue
+        try:
+            lst = list(outs)
+        except Exception:
+            continue
+        changed = False
+        for i, ov in enumerate(lst):
+            if (old_v is not None and ov is old_v) or (
+                old_name and _v_name(ov) == old_name
+            ):
+                lst[i] = new_v
+                changed = True
+        if changed:
+            try:
+                setattr(graph, attr, lst)
+            except Exception:
+                # last resort: rename
+                for i, ov in enumerate(getattr(graph, attr, [])):
+                    if (old_v is not None and ov is old_v) or (
+                        old_name and _v_name(ov) == old_name
+                    ):
+                        try:
+                            ov.name = _v_name(new_v)
+                        except Exception:
+                            pass
+
+
+def _build_use_maps(nodes: List["ir.Node"]):
+    prod_by_obj: Dict[int, int] = {}
+    prod_by_name: Dict[str, int] = {}
+    cons_by_obj: Dict[int, Set[int]] = defaultdict(set)
+    cons_by_name: Dict[str, Set[int]] = defaultdict(set)
+    for i, n in enumerate(nodes):
+        for ov in _node_outputs(n):
+            if ov is None:
+                continue
+            prod_by_obj[id(ov)] = i
+            nm = _v_name(ov)
+            if nm:
+                prod_by_name[nm] = i
+    for i, n in enumerate(nodes):
+        for iv in _node_inputs(n):
+            if iv is None:
+                continue
+            cons_by_obj[id(iv)].add(i)
+            nm = _v_name(iv)
+            if nm:
+                cons_by_name[nm].add(i)
+    return prod_by_obj, prod_by_name, cons_by_obj, cons_by_name
+
+
+def _unique_consumer(
```

Comment:
`value.consumers()` should be sufficient. Essentially, there is no need to maintain a map yourself